core/src/Cuda/Kokkos_Cuda_GraphNodeKernel.hpp

```diff
@@ -34,6 +34,35 @@
 namespace Kokkos {
 namespace Impl {
 
+template <typename Functor>
+struct GraphNodeCaptureImpl<Kokkos::Cuda, Functor> {
```

Comment:
Should this be moved to separate file? Such as `core/src/Cuda/Kokkos_Cuda_GraphNodeCaptureImpl.hpp`? (Just like we had e.g. `Kokkos_GraphNodeThenImpl.hpp`)

Reply:
I'm not sure it's worth adding 3 new files (one for each backend). It's true that the file's name has `NodeKernel` in it. I would wait for other opinions.